### PR TITLE
Upgrade clickhouse jdbc driver

### PR DESCRIPTION
**Pull request by simsicon**
## Purpose of the pull request

Upgrade clickhouse jdbc driver from the deprecated version 0.1.52 to the recommended latest 0.3.2-patch11

According to [the official announcement ](https://github.com/ClickHouse/clickhouse-jdbc)of clickhouse jdbc driver, `ru.yandex.clickhouse.ClickHouseDriver` is now considered deprecated, It's highly recommended to upgrade to 0.3.2+ now for improved performance and stability.

I am also experiencing problems (params not working, socket timeout etc...) with the legacy driver, really hope it could be get upgraded.

## Brief change log

- Change `ru.yandex.clickhouse.ClickHouseDriver 0.1.52` to `com.clickhouse.jdbc.ClickHouseDriver 0.3.2-patch11`

## Verify this pull request

Not sure how to test it, but I have run the build, it looks good to me.

### CODE REVIEW

**Review by kezhenxu94 — DISMISSED** [review]
LGTM if CI passed

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/constants/DataSourceConstants.java
Comment (wemysschen) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/constants/DataSourceConstants.java:31]: are u sure using com.xxx prefixed driver can run clickhouse job normally?

**Comment by simsicon** [comment]
@kezhenxu94 OWASP check failed, does it mean this upgrade has introduced the vulnerabilities to the project?

**Comment by kezhenxu94** [comment]
> @kezhenxu94 OWASP check failed, does it mean this upgrade has introduced the vulnerabilities to the project?

Just ignore that check, it doesn't block the PR from merging if everything else is OK

**Comment by ruanwenjun** [comment]
@simsicon Please check the dependency, you miss some third-part jar.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/constants/DataSourceConstants.java
Comment (simsicon) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/constants/DataSourceConstants.java:31]: yes, `ru.yandex.clickhouse` is deprecated, `com.clickhouse` is the official successor, please refers to https://github.com/ClickHouse/clickhouse-jdbc#clickhouse-java-client--jdbc-driver

**Comment by simsicon** [comment]
> @simsicon Please check the dependency, you miss some third-part jar.

Thanks for the mention, but how can I checkout which dependencies were missed?

**Comment by kezhenxu94** [comment]
> > @simsicon Please check the dependency, you miss some third-part jar.
> 
> Thanks for the mention, but how can I checkout which dependencies were missed?

https://github.com/apache/dolphinscheduler/actions/runs/3368183179/jobs/5600294720#step:7:933


image

**Comment by simsicon** [comment]
@kezhenxu94 thanks for the information, the build is ok now as I can see.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**
